pkg/controller/limited-support/limited_support_controller.go

```diff
@@ -0,0 +1,110 @@
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileLimitedSupportConfigMap{
+		client:            mgr.GetClient(),
+		scheme:            mgr.GetScheme(),
+		metricsAggregator: metrics.GetMetricsAggregator(),
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("limited-support-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource limited-support ConfigMap
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, predicate.Funcs{
+		CreateFunc: func(createEvent event.CreateEvent) bool {
+			return createEvent.Meta.GetName() == limitedSupportConfigMapName && createEvent.Meta.GetNamespace() == limitedSupportConfigMapNamespace
+
+		},
+		DeleteFunc: func(deleteEvent event.DeleteEvent) bool {
+			return deleteEvent.Meta.GetName() == limitedSupportConfigMapName && deleteEvent.Meta.GetNamespace() == limitedSupportConfigMapNamespace
+		},
+		UpdateFunc: func(updateEvent event.UpdateEvent) bool {
+			return updateEvent.MetaNew.GetName() == limitedSupportConfigMapName && updateEvent.MetaNew.GetNamespace() == limitedSupportConfigMapNamespace
+		},
+		GenericFunc: func(genericEvent event.GenericEvent) bool {
+			return genericEvent.Meta.GetName() == limitedSupportConfigMapName && genericEvent.Meta.GetNamespace() == limitedSupportConfigMapNamespace
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileGroup implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileLimitedSupportConfigMap{}
+
+// ReconcileLimitedSupportConfigMap reconciles a ConfigMap object
+type ReconcileLimitedSupportConfigMap struct {
+	client            client.Client
+	scheme            *runtime.Scheme
+	metricsAggregator *metrics.AdoptionMetricsAggregator
+}
+
+// Reconcile reads that state of the cluster for a ConfigMap object limited-support and makes changes based the contained data
+func (r *ReconcileLimitedSupportConfigMap) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling Limited Support ConfigMap")
+
+	// Fetch the ConfigMap openshift-osd-metrics/limited-support
+	cfgMap := &corev1.ConfigMap{}
+	ns := limitedSupportConfigMapNamespace
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: ns, Name: limitedSupportConfigMapName}, cfgMap)
+	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
Should this set the metric to false?

Reply:
Good catch. I will change it. I accidentally left it in when I copied it from the configmap_controller